client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/operations/ReplaceIfUnmodifiedOperation.java

```diff
@@ -35,6 +36,7 @@ protected VersionedOperationResponse executeOperation(Transport transport) {
       transport.writeArray(key);
       transport.writeVInt(lifespan);
       transport.writeVInt(maxIdle);
+      codec.writeExpirationNanoTimes(transport, lifespanNanos, maxIdleNanos, internalFlags);
```

Comment:
@isavin My suggestion: since you're changing the protocol version anyway, don't use any flags, just write the lifespan and maxIdle in millis as vLongs. You can use nanos instead if you prefer, but I doubt we will provide sub-millisecond precision for expiration any time soon (even in embedded mode). 
I would also remove the special case for > 30 days, I'm pretty sure the JCache spec doesn't have any limits for the lifespan...